Filter/BooleanFilter.php

```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                     continue;
                 }
 
-                $values[] = ($v == BooleanType::TYPE_YES) ? 1 : 0;
+                $values[] = ($v == BooleanType::TYPE_YES) ? true : false;
```

Comment:
It is not the same, 
BooleanType::TYPE_YES = 1 and BooleanType::TYPE_YES = 2
https://github.com/sonata-project/SonataCoreBundle/blob/master/Form/Type/BooleanType.php